src/main/java/core/basesyntax/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.web.filter;
+
+import core.basesyntax.lib.Injector;
+import core.basesyntax.service.DriverService;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driverId";
+    private static final Injector INJECTOR = Injector.getInstance("core.basesyntax");
+    private final DriverService driverService = (DriverService) INJECTOR
+            .getInstance(DriverService.class);
+    private final Set<String> urls = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        urls.add("/login");
+        urls.add("/drivers/create");
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String url = req.getServletPath();
+        if (urls.contains(url)) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        if (driverId == null || driverService.get(driverId) == null) {
```

Comment:
can service return null?

Reply:
only in case when user delete himself :) But I suggest, that we will not have such edge situation, am I right?)

Reply:
fixed